services/cd-service/pkg/repository/transformer.go

```diff
@@ -641,6 +689,28 @@ func (u *UndeployApplication) Transform(ctx context.Context, state *State) (stri
 
 	}
 	// remove application
+	releasesDir := fs.Join(appDir, "releases")
+	files, err := fs.ReadDir(releasesDir)
+	if err != nil {
+		return "", nil, fmt.Errorf("could not read the releases directory: %w", err)
```

Comment:
Please add "releasesDir" to the error message:
```suggestion
		return "", nil, fmt.Errorf("could not read the releases directory %s %w", releasesDir, err)
```